kernel/thread.c

```diff
@@ -1010,6 +1015,32 @@ void dump_all_threads(void) {
     THREAD_UNLOCK(state);
 }
 
+#if THREAD_STATS
+void dump_threads_stats(void) {
+    thread_t *t;
+
+    THREAD_LOCK(state);
+    list_for_every_entry (&thread_list, t, thread_t, thread_list_node) {
+        if (t->magic != THREAD_MAGIC) {
+            dprintf(INFO, "bad magic on thread struct %p, aborting.\n", t);
+            hexdump(t, sizeof(thread_t));
+            break;
+        }
+        if (thread_is_idle(t)) {
+            continue;
+        }
+        // thread specific stats
+        dprintf(INFO, "\t(%s):\n", t->name);
```

Comment:
Would also probably make sense to dump a per thread version of this inside dump_thread so you see it when you run the `threads` command.